src/link/analytics-metadata/interfaces.ts

```diff
@@ -0,0 +1,23 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+interface GeneratedAnalyticsMetadataLinkClick {
+  action: 'click';
+  detail: {
+    label: string;
+    href: string;
+    external: string;
+  };
+}
+
+interface GeneratedAnalyticsMetadataLinkComponent {
+  name: 'awsui.Link';
+  label: string;
+  properties: {
+    variant: string;
+  };
+}
+
+export interface GeneratedAnalyticsMetadataLinkFragment extends Partial<GeneratedAnalyticsMetadataLinkClick> {
```

Comment:
What is the point of this `Partial`? All fields are set anyway

Reply:
Removed